### PR TITLE
371 fix format for dates/times

### LINKED ISSUE #371: format.teal_slices cuts the content after `:` if an object is a date for trim_lines = FALSE
### What happened?

```{R}
x <- 
  filter_var(
    dataname = "ADSL", 
    varname = "COUNTRY", 
    selected = c("2023-06-29 18:05:32 CEST", "2023-06-29 18:15:32 CEST"), 
    fixed = TRUE
  )

y <- filter_settings(x)
cat(format(y, trim_lines = FALSE))
{
  "slices": [
    {
      "dataname" : "ADSL",
      "varname"  : "COUNTRY",
      "selected" : ["2023-06-29 18
      "fixed"    : true,
      "locked"   : false,
      "multiple" : true,
      "id"       : "ADSL COUNTRY"
    }
  ],
  "attributes": {
    "module_add" : true
  }
}
```
however it should be `"2023-06-29 18:05:32 CEST", "2023-06-29 18:15:32 CEST"` for the selected field

### sessionInfo()

```R
R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] teal.slice_0.3.0.9005 testthat_3.1.8        shiny_1.7.4          

loaded via a namespace (and not attached):
 [1] httr_1.4.6                sass_0.4.6               
 [3] pkgload_1.3.2             tidyr_1.3.0              
 [5] jsonlite_1.8.4            viridisLite_0.4.2        
 [7] bslib_0.4.2               brio_1.1.3               
 [9] yaml_2.3.7                remotes_2.4.2            
[11] sessioninfo_1.2.2         formatters_0.5.0.9001    
[13] pillar_1.9.0              backports_1.4.1          
[15] glue_1.6.2                teal.widgets_0.3.0.9001  
[17] digest_0.6.31             promises_1.2.0.1         
[19] checkmate_2.2.0           colorspace_2.1-0         
[21] shinycssloaders_1.0.0     htmltools_0.5.5          
[23] httpuv_1.6.11             pkgconfig_2.0.3          
[25] devtools_2.4.5            logger_0.2.2             
[27] teal.transform_0.3.0.9001 purrr_1.0.1              
[29] xtable_1.8-4              scales_1.2.1             
[31] processx_3.8.1            later_1.3.1              
[33] tibble_3.2.1              generics_0.1.3           
[35] ggplot2_3.4.2             usethis_2.1.6            
[37] ellipsis_0.3.2            cachem_1.0.8             
[39] withr_2.5.0               shinyjs_2.1.0            
[41] lazyeval_0.2.2            cli_3.6.1                
[43] magrittr_2.0.3            crayon_1.5.2             
[45] mime_0.12                 memoise_2.0.1            
[47] ps_1.7.5                  fs_1.6.2                 
[49] fansi_1.0.4               teal_0.13.0.9004         
[51] pkgbuild_1.4.0            profvis_0.3.8            
[53] tools_4.2.3               data.table_1.14.8        
[55] prettyunits_1.1.1         lifecycle_1.0.3          
[57] stringr_1.5.0             plotly_4.10.2            
[59] munsell_0.5.0             callr_3.7.3              
[61] compiler_4.2.3            jquerylib_0.1.4          
[63] rlang_1.1.1               grid_4.2.3               
[65] rstudioapi_0.14           htmlwidgets_1.6.2        
[67] miniUI_0.1.1.1            shinyWidgets_0.7.6       
[69] gtable_0.3.3              teal.logger_0.1.1.9006   
[71] teal.data_0.2.0.9002      R6_2.5.1                 
[73] dplyr_1.1.2               fastmap_1.1.1            
[75] utf8_1.2.3                rprojroot_2.0.3          
[77] desc_1.4.2                stringi_1.7.12           
[79] Rcpp_1.0.10               vctrs_0.6.2              
[81] tidyselect_1.2.0          urlchecker_1.0.1
```

### Relevant log output

```R
teal.slice@299_...@filter_panel_refactor@main
teal@filter_panel_refactor@main
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by m7pr**
fix #371

### CODE REVIEW

File: R/teal_slice.R [hunk 1]
```diff
@@ -586,14 +586,14 @@ justify_json <- function(json) {
   }
 
   json_lines <- strsplit(json, "\n")[[1]]
-  json_lines_split <- strsplit(json_lines, split = ":", fixed = TRUE)
-  name_width <- max(unlist(gregexpr(":", json_lines))) - 1
+  json_lines_split <- regmatches(json_lines, regexpr(":", json_lines), invert = TRUE)
```
Comment (gogonzo) [inline R/teal_slice.R:589]:
```suggestion
  json_lines_split <- strsplit(json_lines, split = "^([^,]+),", fixed = TRUE)
```

Comment (m7pr) [reply R/teal_slice.R:589]: @gogonzo I don't get what is your change doing here? you do not have `:` in the split regex
this is result of your change
`````{R}
> strsplit('"value": "fields with dates 02:04:03"', split = "^([^,]+),", fixed = TRUE)
[[1]]
[1] "\"value\": \"fields with dates 02:04:03\""
````
We actually wanna split by the first appearance of `:`

```{R}
regmatches('"value": "fields with dates 02:04:03"', regexpr(":", '"value": "fields with dates 02:04:03"'), invert = TRUE)
[[1]]
[1] "\"value\""                       " \"fields with dates 02:04:03\""
```

**Comment by m7pr** [comment]
@gogonzo the examples that take too long are in FilteredData.R
https://github.com/insightsengineering/teal.slice/actions/runs/5414514729/jobs/9841689850?pr=372#step:20:113

**Comment by chlebowa** [comment]
> @gogonzo the examples that take too long are in FilteredData.R https://github.com/insightsengineering/teal.slice/actions/runs/5414514729/jobs/9841689850?pr=372#step:20:113
> 
> We might add `\notrun{}` in a couple of places but I don't see any good solution and I don't even think 5seconds is the proper threshold for examples

`notrun` is ignored on CRAN anyway.

File: R/teal_slice.R (hunk 1 above)
Comment (gogonzo) [reply R/teal_slice.R:589]: Sorry, replace `,` with `:`

**Review by gogonzo — APPROVED** [review]
👍 format is ok

```
filter_var(
  dataname = "x:x", 
  varname = "y     ::", 
  choices = c("x:x", "y:y:y", "cjkdb.,f7"),
  selected = "x:x"
) |> print(trim_lines = FALSE)

# {
#  "dataname" : "x:x",
#  "varname"  : "y     ::",
#  "choices"  : ["x:x", "y:y:y", "cjkdb.,f7"],
#  "selected" : ["x:x"],
#  "fixed"    : false,
#  "locked"   : false,
#  "multiple" : true,
#  "id"       : "x:x y     ::"
#}
```

**Comment by m7pr** [comment]
I thought I also provided an exmaple

```{R}
x <- 
  filter_var(
    dataname = "ADSL", 
    varname = "COUNTRY", 
    selected = as.POSIXct(c("2023-06-29 18:05:32 CEST", "2023-06-29 18:15:32 CEST")), 
    fixed = TRUE
  )

cat(format(x, trim_lines = FALSE))
```

output


```{R}
{
  "dataname" : "ADSL",
  "varname"  : "COUNTRY",
  "selected" : ["2023-06-29 18:05:32", "2023-06-29 18:15:32"],
  "fixed"    : true,
  "locked"   : false,
  "multiple" : true,
  "id"       : "ADSL COUNTRY"
}
```

**Pull request merged**